include/maidsafe/crux/socket.hpp

```diff
@@ -253,34 +254,32 @@ inline void socket::close() {
     keepalive_timer.stop();
     transmit_queue.shutdown();
 
+    auto m = multiplexer;
```

Comment:
Shouldn't the check for null be done on the pinned copy (Thanks for the term "pin" by the way @breese) otherwise the multiplexer could be destroyed between the point where we check and the point where we pin it?